src/EventGenerator/EmitEvent.php

```
@@ -132,6 +145,11 @@ public static function create(ContainerInterface $container, array $configuratio
   public function execute($entity = NULL) {
     // Generate event as stomp message.
     try {
+      if (is_null($this->stomp->getClient()->getProtocol())) {
```

Fairly nit but maybe toss a comment in here that references the [created issue](https://github.com/stomp-php/stomp-php/issues/167) in the event (heh) things get fixed upstream? May be worth maybe noting that the `NULL` is expected as per https://github.com/stomp-php/stomp-php/blob/3a9347a11743d0b79fd60564f356bc3efe40e615/src/Client.php#L429-L434. Lastly, while you mentioned it in the issue perhaps just noting that the odd `disconnect` + `connect` behavior is purely meant to throw an exception if things are actually broken/not configured.

EDIT: If the intent is to just throw an exception when it's `NULL` should we just throw the exception directly? If that's the case I feel like that's somewhat what @adam-vessey was leaning towards on the tech call in regards to having the `islandora.stomp` service itself blow up because it's in an unexpected state (heh).

---

I see what you're saying. I was (maybe naively) hoping that if the JMS service was still up this might restore your connection. But based on what I understand to be the lifecycle of the client, it is probably unlikely to bounce in a single request. So I can just stick the logging/messenger stuff in here and `return;`

---

I don't disagree with the retry attempt, just was confirming the expectations was all!

---

I looked at how the to have the stomp service kill itself, but it was not immediately apparent and so my mind went to a new service to wrap the StatefulStomp client. That seemed like overkill. I'll add a comments requested at the disconnect + connect.